gnomad_qc/v4/assessment/calculate_per_sample_stats.py

```diff
@@ -154,13 +154,24 @@ def create_per_sample_counts_ht(
         filter_expr["non_coding"] = filter_expr["pass_filters"] & hl.if_else(
             hl.any(lambda csq: mt.most_severe_csq == csq, CSQ_NON_CODING), True, False
         )
-        filter_expr["lof"] = (
-            filter_expr["pass_filters"]
-            & hl.if_else(
-                hl.any(lambda csq: mt.most_severe_csq == csq, LOF_CSQ_SET), True, False
-            )
-            & (mt.lof == "HC")
-            & mt.no_lof_flags
+        filter_expr["lof"] = filter_expr["pass_filters"] & hl.if_else(
+            hl.any(lambda csq: mt.most_severe_csq == csq, LOF_CSQ_SET), True, False
+        )
+        filter_expr["lof_HC"] = filter_expr["lof"] & (mt.lof == "HC")
+        filter_expr["lof_LC"] = filter_expr["lof"] & (mt.lof == "LC")
+        filter_expr["lof_OS"] = filter_expr["lof"] & (mt.lof == "OS")
+        filter_expr["lof_HC_no_flags"] = filter_expr["lof_HC"] & mt.no_lof_flags
+        filter_expr["splice_acceptor_variant"] = filter_expr["lof_HC_no_flags"] & (
+            mt.most_severe_csq == "splice_acceptor_variant"
+        )
+        filter_expr["splice_donor_variant"] = filter_expr["lof_HC_no_flags"] & (
+            mt.most_severe_csq == "splice_donor_variant"
+        )
+        filter_expr["stop_gained"] = filter_expr["lof_HC_no_flags"] & (
+            mt.most_severe_csq == "stop_gained"
+        )
+        filter_expr["frameshift_variant"] = filter_expr["lof_HC_no_flags"] & (
+            mt.most_severe_csq == "frameshift_variant"
```

Comment:
I believe we want this breakdown in _LC and _OS and _HC_with_flags as well!

Reply:
world's laziest implementation - you may have already written something smarter !! : 

```
for lof_label in ["lof_HC","lof_LC","LOF_OS","LOF_HC_no_flags"]:

    filter_expr[f"{lof_label}_splice_acceptor_variant"] = filter_expr[lof_label] & (
        mt.most_severe_csq == "splice_acceptor_variant"
    )
    filter_expr[f"{lof_label}_splice_donor_variant"] = filter_expr[lof_label] & (
        mt.most_severe_csq == "splice_donor_variant"
    )
    filter_expr[f"{lof_label}_stop_gained"] = filter_expr[lof_label] & (
        mt.most_severe_csq == "stop_gained"
    )
    filter_expr[f"{lof_label}_frameshift_variant"] = filter_expr[lof_label] & (
        mt.most_severe_csq == "frameshift_variant"
```

Reply:
Ahhhh, that's really too much...but I will ask chatGPT to simplify a bit

Reply:
too much for what? Too many lines and inefficient writing , or too many aggregations to calculate ? 

I won't be offended if it's the prior, I wrote this up in like two minutes just as an example

Reply:
No, I meant, it will came out with a lot of results...not your code.

Reply:
Sorry for not being clear. I'm still waiting for KC to comment, you can comment in the [sheet](https://docs.google.com/spreadsheets/d/1bX0pvBvbNalh-_VS-IuHrX-mv0MjJe_pWYMjRoC6Rto/edit?usp=sharing) and I will improve the code once we got the final computation we need to run:

Reply:
Ah sorry - this was actually KC's idea! I had messaged her about it before suggesting this change, and she said that it would be helpful to have breakdowns within each LOFTEE category, to potentially explain why behavior is different (if it even is). I should've let you know a bit beforehand that this was the case